components/builder-web/app/actions/packages.ts

```diff
@@ -68,12 +78,38 @@ export function fetchPackage(pkg) {
         dispatch(clearPackages());
         depotApi.get(pkg.ident).then(response => {
             dispatch(setCurrentPackage(response["results"]));
+            dispatch(fetchCurrentPackageChannels(pkg));
         }).catch(error => {
             dispatch(setCurrentPackage(undefined, error));
         });
     };
 }
 
+export function fetchCurrentPackageChannels(pkg) {
+    return dispatch => {
+        dispatch(clearPackageChannels());
+        depotApi.getPackageChannels(pkg.ident)
+            .then(channels => dispatch(setCurrentPackageChannels(channels)))
+            .catch(error => dispatch(setCurrentPackageChannels(undefined, error)));
+    };
+}
+
+export function fetchVisiblePackageChannels(pkgs) {
+    return dispatch => {
+        pkgs.forEach((pkg) => {
+
+            // Only fetch for fully qualified packages
+            if (pkg.origin && pkg.name && pkg.version && pkg.release) {
+                depotApi.getPackageChannels(pkg)
+                    .then(channels => {
+                        dispatch(setVisiblePackageChannels(pkg, channels));
+                    })
+                .catch(error => console.error(error));
+            }
+        });
+    };
+}
```

Comment:
Using the existing APIs, this is definitely a good way to accomplish the goal of retrieving a list of channels for every package.  I'm not opposed to this at all.  I do think it _might_ be possible to accomplish your goal without having the overhead of making an HTTP request for every package.  I'm going to try and spike out a quick POC for what I mean.